push to docker workflow fails in test phase

**Describe the bug**
The docker push workflow fails in the test phase after the push

```txt
Run echo "Install container tools"
  echo "Install container tools"
  curl -LO https://storage.googleapis.com/container-structure-test/latest/container-structure-test-linux-amd64 \
    && chmod +x container-structure-test-linux-amd64 \
    && sudo mv container-structure-test-linux-amd64 /usr/local/bin/container-structure-test
  echo "Run tests"
  container-structure-test test \
    --image ghcr.io/andreasaugustin/actions-template-sync:main \
    --pull --config src/test-config.yaml
  shell: /usr/bin/bash -e {0}
Install container tools
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 14.3M  100 14.3M    0     0  3[2](https://github.com/AndreasAugustin/actions-template-sync/actions/runs/3623570253/jobs/6109588041#step:8:2).7M      0 --:--:-- --:--:-- --:--:-- [3](https://github.com/AndreasAugustin/actions-template-sync/actions/runs/3623570253/jobs/6109588041#step:8:3)2.7M
Run tests
time="2022-12-05T20:07:06Z" level=fatal msg="error pulling remote image ghcr.io/andreasaugustin/actions-template-sync:main: API error ([4](https://github.com/AndreasAugustin/actions-template-sync/actions/runs/3623570253/jobs/6109588041#step:8:4)04): pull access denied for andreasaugustin/actions-template-sync, repository does not exist or may require 'docker login': denied: requested access to the resource is denied"
Error: Process completed with exit code 1.
```

**To Reproduce**

When the github action kicks in (on push to some branches) the action fails in last step

**Expected behavior**
The test does not fail


**Screenshots**

![image](https://user-images.githubusercontent.com/8027933/205734303-da813275-1f2a-481e-a30d-f3a76a7537ce.png)


 **Additional context**
The push itself is working. Maybe it is a timing related issue

--- a/.github/workflows/push_docker.yml
+++ b/.github/workflows/push_docker.yml
@@ -6,7 +6,7 @@ on:
   # Also run on updates to this repo.
   push:
     branches:
-      - fix/*
+      - main
     tags:
       - 'v*'
     paths-ignore:

--- a/.github/workflows/push_docker.yml
+++ b/.github/workflows/push_docker.yml
@@ -60,7 +60,8 @@ jobs:
           push: false
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
-      name: run tests
+      -
+        name: run tests
         run: |
           echo "Install container tools"
           curl -LO https://storage.googleapis.com/container-structure-test/latest/container-structure-test-linux-amd64 \

--- a/.github/workflows/push_docker.yml
+++ b/.github/workflows/push_docker.yml
@@ -87,14 +87,14 @@ jobs:
       #     repository: andyAugustin/${{ matrix.variant }}
       #     # readme-filepath: ./images/${{ matrix.variant }}/README.md
       #     # short-description: See https://github.com/${{ github.repository }}/images/${{ matrix.variant }} for the source.
-      -
-        name: run tests
-        run: |
-          echo "Install container tools"
-          curl -LO https://storage.googleapis.com/container-structure-test/latest/container-structure-test-linux-amd64 \
-            && chmod +x container-structure-test-linux-amd64 \
-            && sudo mv container-structure-test-linux-amd64 /usr/local/bin/container-structure-test
-          echo "Run tests"
-          container-structure-test test \
-            --image ghcr.io/andreasaugustin/${{ matrix.variant }}:main \
-            --pull --config src/test-config.yaml
+      # -
+      #   name: run tests
+      #   run: |
+      #     echo "Install container tools"
+      #     curl -LO https://storage.googleapis.com/container-structure-test/latest/container-structure-test-linux-amd64 \
+      #       && chmod +x container-structure-test-linux-amd64 \
+      #       && sudo mv container-structure-test-linux-amd64 /usr/local/bin/container-structure-test
+      #     echo "Run tests"
+      #     container-structure-test test \
+      #       --image ghcr.io/andreasaugustin/${{ matrix.variant }}:main \
+      #       --pull --config src/test-config.yaml

--- a/.github/workflows/push_docker.yml
+++ b/.github/workflows/push_docker.yml
@@ -52,14 +52,22 @@ jobs:
             andyAugustin/${{ matrix.variant }}
             ghcr.io/AndreasAugustin/${{ matrix.variant }}
       -
-        name: Build docker ${{ matrix.variant }}
-        id: build_docker
+        name: Build and push ${{ matrix.variant }}
         uses: docker/build-push-action@v3
         with:
           context: ./src/
-          push: false
+          push: ${{ github.event_name != 'pull_request' }}
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
+      -
+        name: Docker Hub Description
+        uses: peter-evans/dockerhub-description@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+          repository: andyAugustin/${{ matrix.variant }}
+          # readme-filepath: ./images/${{ matrix.variant }}/README.md
+          # short-description: See https://github.com/${{ github.repository }}/images/${{ matrix.variant }} for the source.
       -
         name: run tests
         run: |
@@ -71,31 +79,3 @@ jobs:
           container-structure-test test \
             --image ghcr.io/andreasaugustin/${{ matrix.variant }}:${{ steps.meta.outputs.version }} \
             --config src/test-config.yaml
-      # -
-      #   name: Build and push ${{ matrix.variant }}
-      #   uses: docker/build-push-action@v3
-      #   with:
-      #     context: ./src/
-      #     push: ${{ github.event_name != 'pull_request' }}
-      #     tags: ${{ steps.meta.outputs.tags }}
-      #     labels: ${{ steps.meta.outputs.labels }}
-      # -
-      #   name: Docker Hub Description
-      #   uses: peter-evans/dockerhub-description@v3
-      #   with:
-      #     username: ${{ secrets.DOCKERHUB_USERNAME }}
-      #     password: ${{ secrets.DOCKERHUB_TOKEN }}
-      #     repository: andyAugustin/${{ matrix.variant }}
-      #     # readme-filepath: ./images/${{ matrix.variant }}/README.md
-      #     # short-description: See https://github.com/${{ github.repository }}/images/${{ matrix.variant }} for the source.
-      # -
-      #   name: run tests
-      #   run: |
-      #     echo "Install container tools"
-      #     curl -LO https://storage.googleapis.com/container-structure-test/latest/container-structure-test-linux-amd64 \
-      #       && chmod +x container-structure-test-linux-amd64 \
-      #       && sudo mv container-structure-test-linux-amd64 /usr/local/bin/container-structure-test
-      #     echo "Run tests"
-      #     container-structure-test test \
-      #       --image ghcr.io/andreasaugustin/${{ matrix.variant }}:main \
-      #       --pull --config src/test-config.yaml

--- a/.github/workflows/push_docker.yml
+++ b/.github/workflows/push_docker.yml
@@ -69,7 +69,7 @@ jobs:
             && sudo mv container-structure-test-linux-amd64 /usr/local/bin/container-structure-test
           echo "Run tests"
           container-structure-test test \
-            --image ghcr.io/andreasaugustin/${{ matrix.variant }}:${{ steps.meta.outputs.version }}
+            --image ghcr.io/andreasaugustin/${{ matrix.variant }}:${{ steps.meta.outputs.version }} \
             --config src/test-config.yaml
       # -
       #   name: Build and push ${{ matrix.variant }}

--- a/.github/workflows/push_docker.yml
+++ b/.github/workflows/push_docker.yml
@@ -69,7 +69,7 @@ jobs:
             && sudo mv container-structure-test-linux-amd64 /usr/local/bin/container-structure-test
           echo "Run tests"
           container-structure-test test \
-            --image ${{ steps.build_docker.outputs.imageid }}
+            --image ghcr.io/andreasaugustin/${{ matrix.variant }}/${{ steps.meta.outputs.version }}
             --config src/test-config.yaml
       # -
       #   name: Build and push ${{ matrix.variant }}

--- a/.github/workflows/push_docker.yml
+++ b/.github/workflows/push_docker.yml
@@ -6,7 +6,7 @@ on:
   # Also run on updates to this repo.
   push:
     branches:
-      - main
+      - fix/*
     tags:
       - 'v*'
     paths-ignore:
@@ -52,22 +52,41 @@ jobs:
             andyAugustin/${{ matrix.variant }}
             ghcr.io/AndreasAugustin/${{ matrix.variant }}
       -
-        name: Build and push ${{ matrix.variant }}
+        name: Build docker ${{ matrix.variant }}
+        id: build_docker
         uses: docker/build-push-action@v3
         with:
           context: ./src/
-          push: ${{ github.event_name != 'pull_request' }}
+          push: false
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
-      -
-        name: Docker Hub Description
-        uses: peter-evans/dockerhub-description@v3
-        with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
-          repository: andyAugustin/${{ matrix.variant }}
-          # readme-filepath: ./images/${{ matrix.variant }}/README.md
-          # short-description: See https://github.com/${{ github.repository }}/images/${{ matrix.variant }} for the source.
+      name: run tests
+        run: |
+          echo "Install container tools"
+          curl -LO https://storage.googleapis.com/container-structure-test/latest/container-structure-test-linux-amd64 \
+            && chmod +x container-structure-test-linux-amd64 \
+            && sudo mv container-structure-test-linux-amd64 /usr/local/bin/container-structure-test
+          echo "Run tests"
+          container-structure-test test \
+            --image ${{ steps.build_docker.outputs.imageid }}
+            --config src/test-config.yaml
+      # -
+      #   name: Build and push ${{ matrix.variant }}
+      #   uses: docker/build-push-action@v3
+      #   with:
+      #     context: ./src/
+      #     push: ${{ github.event_name != 'pull_request' }}
+      #     tags: ${{ steps.meta.outputs.tags }}
+      #     labels: ${{ steps.meta.outputs.labels }}
+      # -
+      #   name: Docker Hub Description
+      #   uses: peter-evans/dockerhub-description@v3
+      #   with:
+      #     username: ${{ secrets.DOCKERHUB_USERNAME }}
+      #     password: ${{ secrets.DOCKERHUB_TOKEN }}
+      #     repository: andyAugustin/${{ matrix.variant }}
+      #     # readme-filepath: ./images/${{ matrix.variant }}/README.md
+      #     # short-description: See https://github.com/${{ github.repository }}/images/${{ matrix.variant }} for the source.
       -
         name: run tests
         run: |

--- a/.github/workflows/push_docker.yml
+++ b/.github/workflows/push_docker.yml
@@ -69,7 +69,7 @@ jobs:
             && sudo mv container-structure-test-linux-amd64 /usr/local/bin/container-structure-test
           echo "Run tests"
           container-structure-test test \
-            --image ghcr.io/andreasaugustin/${{ matrix.variant }}/${{ steps.meta.outputs.version }}
+            --image ghcr.io/andreasaugustin/${{ matrix.variant }}:${{ steps.meta.outputs.version }}
             --config src/test-config.yaml
       # -
       #   name: Build and push ${{ matrix.variant }}

--- a/README.md
+++ b/README.md
@@ -63,8 +63,7 @@ You will receive a pull request within your repository if there are some changes
 | github_token | Token for the repo. Can be passed in using `$\{{ secrets.GITHUB_TOKEN }}` | `true` |  |
 | source_repo_path | Repository path of the template | `true` | |
 | upstream_branch | The target branch | `true` | `main` |
-| source_repo_ssh_private_key | `[optional]` private ssh key for the source repository. E.q. useful if using a private template repository.
-[see](#private-template-repository)| `false` |  |
+| source_repo_ssh_private_key | `[optional]` private ssh key for the source repository. [see](#private-template-repository)| `false` |  |
 | pr_branch_name_prefix | `[optional]` the prefix of branches created by this action | `false` | `chore/template_sync`  |
 | pr_title | `[optional]` the title of PRs opened by this action. Must be already created. | `false` | `upstream merge template repository`  |
 | pr_labels | `[optional]` comma separated list. [pull request labels][pr-labels]. Must be already created. | `false` | |

